src/main/java/com/spotify/sparkey/IndexHash.java

```diff
@@ -89,7 +87,7 @@ static IndexHash open(File indexFile, File logFile) throws IOException {
       indexHash = new IndexHash(indexFile, logFile, header, logHeader, indexData, maxBlockSize, logData);
       indexHash.validate();
       return indexHash;
-    } catch (Exception e) {
+    } catch (IOException | RuntimeException | Error e) {
```

Comment:
Why would you catch `Error` here?

Reply:
Just to make sure we can clean up in case something really bad happens

Reply:
If an error happens, the process is going to terminate, and closing file handles can potentially do IO or allocate memory, both of which are not allowed during an error.  Catching this and trying to do cleanup probably does more harm than good.